game-core/src/main/java/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -36,7 +34,7 @@ private ClientFileSystemHelper() {}
    */
   public static File getRootFolder() {
     try {
-      return getFolderContainingFileWithName(GAME_ENGINE_PROPERTIES_FILE, getCodeSourceFolder());
+      return getFolderContainingFileWithName(".triplea-root", getCodeSourceFolder());
```

Comment:
I'm actually really surprised that such a reference file is needed.
I quickly had a look at some usages and from my understanding it seems that this is only being used to locate assets and dice_server folders etc.
Usually `SomeClass.class.getResource("/").toFile()` or something should work good enough to figure out where the jar file is, but I can see that this will probably not work as expected when the class files are deployed or something.
I just looked and it's possible to configure install4j to always provide some CLI arguments to the jar file.
So we could make the location of the assets and dice_servers folder a command line argument and avoid this "hack" as I see it entirely.
Not necessarily as part of this PR, but definitely in the long run.
Also we could think about packing the assets folder into the fat jar, which would drop the necessity to "figure out where the folder is" entirely for those kind of files.

Reply:
> but the "removal" of the file didn't simplify things as much as I hoped

As I'm learning, that's because a lot of this stuff is intertwined, and I think it's all too much to fix in a single PR.  Once the _assets_ folder problem is solved, I expect things will clean up nicely.

> I quickly had a look at some usages and from my understanding it seems that this is only being used to locate assets and dice_server folders etc.

There's two other usages of `getRootFolder()` that really make things frustrating (especially because I don't think one of them [in `ProcessRunnerUtil`] even works).  And, as you noted, trying to support both jar'ed and file system installations makes things even worse.

> So we could make the location of the assets and dice_servers folder a command line argument and avoid this "hack" as I see it entirely.

As long as whatever solution we come up with works when running from a) the IDE, b) the Gradle `run` task, c) the portable build, and d) the installer, I'm ok with it. :smile:  I like the CLI arg idea, as long as we default to the CWD if it's not specified (that would pretty much cover all the non-installer cases).

> Also we could think about packing the assets folder into the fat jar, which would drop the necessity to "figure out where the folder is" entirely for those kind of files.

I considered that, but then I remembered reading some recent posts on the forum where users were talking about customizing the images in the _assets_ folder for various reasons.  While it would still be possible for users to do that if we bundled the assets in the jar, it would be two more steps.  (And it would stop working if we ever decide to sign the jar someday.)  However, I'm not sure how strong an argument this "requirement" is against us moving _assets_ into the jar.